internal/course/database/course_repository.go

```diff
@@ -111,38 +111,14 @@ func (r CourseRepository) UpdateCourseByID(c *domain.Course) error {
 	return nil
 }
 
-func (r CourseRepository) UpdateCourseByCode(c *domain.Course) error {
```

Comment:
I still think it makes sense to have update by code. But also depends how we planned to expose the list of courses.

Reply:
I think that for now, we can help it simple. We can include these extra scenarios later. There would be also the need to get by code, but... let's first achieve the basics.